pkg/3scale/amp/template/adapters/system.go

```diff
@@ -161,6 +165,52 @@ func (s *System) options() (*component.SystemOptions, error) {
 	sob.BackendSharedSecret("${SYSTEM_BACKEND_SHARED_SECRET}")
 	sob.TenantName("${TENANT_NAME}")
 	sob.WildcardDomain("${WILDCARD_DOMAIN}")
-	sob.StorageClassName(nil)
+
+	s.setFileStorageOptions(&sob)
 	return sob.Build()
 }
+
+func (s *System) setFileStorageOptions(sob *component.SystemOptionsBuilder) {
+	switch s.FileStorageType {
+	case component.SystemFileStorageTypePVC:
+		// We set the storageClassName as nil even though the value comes
+		// as a template parameter in this case.
+		// The reason for this is that the PersistentVolumeClaim Kubernetes object
+		// requires a pointer to string to specify the StorageClassName. For this
+		// you need to reference it using OpenShift double-brace parameter
+		// expansion, which does not exist in Kubernetes. We take care of replacing
+		// the value with the double brace parameter expansion as a postProcess
+		// step after generating the template objects
+		sob.PVCFileStorageOptions(component.PVCFileStorageOptions{})
+	case component.SystemFileStorageTypeS3:
+		sob.S3FileStorageOptions(component.S3FileStorageOptions{
+			S3ConfigSecret: "${S3_CONFIGURATION_SECRET_NAME}",
+		})
+	default:
+		panic("s3 or pvc has to be set as system's file storage option")
+	}
+}
+
+func (s *System) fileStorageParameters() []templatev1.Parameter {
+	switch s.FileStorageType {
+	case component.SystemFileStorageTypePVC:
+		return []templatev1.Parameter{
+			templatev1.Parameter{
+				Name:        "RWX_STORAGE_CLASS",
+				Description: "The Storage Class to be used by ReadWriteMany System's PVC",
+				Required:    false,
+				Value:       "null", //TODO this is incorrect. The value that should be used at the end is null and not "null", however template parameters only accept strings.
+			},
+		}
+	case component.SystemFileStorageTypeS3:
+		return []templatev1.Parameter{
+			templatev1.Parameter{
+				Name:        "S3_CONFIGURATION_SECRET_NAME",
```

Comment:
The S3 template now has a new required parameter being the secret name configuration and the old parameters that were the bucket name, region etc... have been removed.
This is an important change on how the s3 template is deployed because now to deploy it it requires the user to pre-create the S3 configuration secret BEFORE deploying the template, whereas before they just filled the S3-specific related values